pyqtgraph/plotDataMappings.py

```diff
@@ -0,0 +1,182 @@
+import warnings
+import numpy as np
+
+__all__ = []
+
+MAPPINGS = {}
+
+def get(name):
+    """ 
+    Returns PlotDataMapping according to specified name
+    """
+    if name not in MAPPINGS:
+        raise KeyError(f"No PlotDataMapping '{name}' has been registered.")
+    return MAPPINGS[name]
+    
+def register(name, mapping):
+    """
+    Registers a new PlotDataMapping object
+    """
+    if not isinstance(mapping, PlotDataMapping):
+        raise TypeError('added object must subclass PlotDataMapping')
+    MAPPINGS[name] = mapping
+    
+
+class PlotDataMapping(object):
```

Comment:
nit-pick level 1000

no need to inherit `object` in modern python3 code, this line can just be `class PlotDataMapping`